Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Generator.cs

```diff
@@ -74,6 +74,20 @@ public void Execute(GeneratorExecutionContext context)
                 foreach (var lambdaMethod in receiver.LambdaMethods)
                 {
                     var lambdaMethodModel = semanticModelProvider.GetMethodSemanticModel(lambdaMethod);
+                    
+                    // Check for necessary references
+                    if (lambdaMethodModel.HasAttribute(context, TypeFullNames.RestApiAttribute)
+                        || lambdaMethodModel.HasAttribute(context, TypeFullNames.HttpApiAttribute))
+                    {
+                        // Check for arbitrary type from "Amazon.Lambda.APIGatewayEvents"
+                        if (context.Compilation.GetTypeByMetadataName(TypeFullNames.APIGatewayProxyRequest) == null)
```

Comment:
In theory it will work but a better check here would be
```suggestion
                        if (context.Compilation.ReferencedAssemblyNames.FirstOrDefault(x => x.Name == "Amazon.Lambda.APIGatewayEvents"))
```

This will check whether the assembly is referenced or not.

Reply:
Ah that's better 🙂